### ISSUE TITLE
Fedora 24: Jasp immediately crashes

### ISSUE THREAD

**Issue by silberzwiebel** [jasp-stats/jasp-desktop#1507]
After finally [managing to compile Jasp under Fedora 24](https://github.com/jasp-stats/jasp-desktop/issues/1430), I started it by running `./jasp`
In the terminal I get these messages:

```
QObject::connect: No such signal FSBModel::hideAuthentication()
QObject::connect: No such signal FSBModel::hideAuthentication()
QObject::connect:  (receiver name: 'browser')
QObject::connect: No such signal FSBMExamples::hideAuthentication()
QLayout: Attempting to add QLayout "" to BackStageWidget "backStage", which already has a layout
QLayout: Attempting to add QLayout "" to VariablesWidget "variablesPage", which already has a layout
```

And as graphical output I see the main Jasp window for a short time, followed by an error dialog that has a non-conclusive error message:

```
JASP has experienced an unexpected internal error.

""

If you could report your experiences to the JASP team that would be appreciated.

JASP cannot continue and will now close.
```

Could you help me to run Jasp?

**Comment by akashrajkn** [comment]
@silberzwiebel, need more details to find what is causing the problem. What version of Qt are you using? [on linux it should be 5.4]

**Comment by akashrajkn** [comment]
Do you have the required R packages at /usr/lib64/R ?

**Comment by silberzwiebel** [comment]
I'm installing them at the moment (and face different R related issues). Apparently, I was to quick to run the program. 

It seems that I'm using Qt 5.6.1.

**Comment by akashrajkn** [comment]
@silberzwiebel , Qt 5.6 does not support webkit module any more. You will have to separately compile this module if you want to use 5.6 - this is a bit of a hassle. It's more convenient to use Qt 5.4

**Comment by silberzwiebel** [comment]
Hmm, okay. I've installed the required R packages but Jasp still won't start.
In the fedora repos there is a QtWebkit package, though: 
```
qt5-qtwebkit.x86_64                     5.6.1-3.b889f46git.fc24
qt5-qtwebkit-devel.x86_64           5.6.1-3.b889f46git.fc24
```
I do have installed both pacakges. There are no Qt 5.4 packages available in the fedora repos. Could you give me a hint for compiling qtwebkit on my own?

**Comment by akashrajkn** [comment]
@silberzwiebel, look at [this link](https://trac.webkit.org/wiki/BuildingQtOnLinux), they tell how to compile webkit

You could download the 5.4 version from here [download.qt](https://download.qt.io/official_releases/qt/5.4/5.4.1/).

**Comment by silberzwiebel** [comment]
I downloaded Qt 5.4 and compiled jasp against it. It still does not work but now it shows a little bit more output:
```
subprocess started
subprocess started
subprocess started
subprocess started
"Fehler in loadNamespace(name) : there is no package called ‘rjson’
"
subprocess error 1
subprocess finished 0
```

After reading this, I installed `rjson` and now both jasps (with Qt 5.4 and 5.6) are working (at least they do not crash immediately.
This issue then might be closed, however, I suggest to fix the empty error message. If I knew a specific R package is the problem, I could install this easier than installing a whole version of Qt amongside the system's Qt. The graphical error message was always empty, for both jasps.
Thanks for the quick help!

**Comment by stephensrmmartin** [comment]
Arch, iirc, uses the latest qt and jasp works fine.

**Comment by akashrajkn** [comment]
@silberzwiebel, I will try to fix that - so JASP gives the correct message when it crashes.

**Comment by boutinb** [comment]
@akashrajkn Thanks for your help Akash!

**Comment by silberzwiebel** [comment]
I tried to compile JASP once more for my second machine that is also running Fedora 24 but unfortunately was not successful. Compiling with the system Qt 5.6 just gives me an empty error message and no hint at the the command line output. Compiling with downloaded Qt 5.4 gives me at least a hint what is going wrong (again something with R...) but I can't get it to work: 
```
subprocess started
subprocess started
subprocess started
subprocess started
"cannot find system Renviron
"
"cannot find system Renviron
"
"cannot find system Renviron
"
"cannot find system Renviron
"
"Fatal error: unable to open the base package

"
"Fatal error: unable to open the base package

"
"Fatal error: unable to open the base package

"
"Fatal error: unable to open the base package

"
subprocess finished 2
subprocess finished 2
subprocess finished 2
subprocess finished 2
```

I've installed all required R packages (as stated [at the bottom of this page](https://github.com/jasp-stats/jasp-desktop/blob/development/Docs/development/jasp-building-guide.md)) inside R into the system-wide library. I've also set the environment variable R_HOME to /usr/lib64/R which is the default location in Fedora. So, running $R_HOME/bin/R and loading all packages results in zero errors.
Any ideas what I could try to make it work?

**Comment by akashrajkn** [comment]
@silberzwiebel, this error pops up when R_HOME cannot be found. In fedora, sometimes, R_HOME could be set to "/usr/local/lib64/R" instead of "/usr/lib64/R". Can you check this?

**Comment by silberzwiebel** [comment]
I do not have the folder /usr/local/lib64/R, setting the system environment variable R_HOME to this (non-existing) path doesn't change JASPs behavior.
Shouldn't setting the R_HOME variable via `export R_HOME=/usr/lib64/R` be sufficient to let JASP find R? Or is it not using this environment variable?

**Comment by silberzwiebel** [comment]
Looked at my problem with my second machine again. It turned out that during my first trouble I created a symlink on the other machine with

```
ln -s /usr/lib64/R /usr/lib/R
```

that is necessary to make JASP find R. Apparently, JASP does not use the environment variable R_HOME at all but only looks at the default path for non-Fedora systems (?). 

Again, I suggest to make better error messages. I only got an idea that something with R was wrong when I compiled JASP with Qt5.4, the version compiled with Qt5.6 is quiet about R errors. Also, it would be nice if JASP could print something like "Looking for R in /usr/lib/R...". If Jasp crashes then, one might have the idea to look at the path to see if it is correct.

Thanks for your work!
